rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -2184,3 +2184,104 @@ rocalPreEmphasisFilter(RocalContext p_context,
+rocalNonSilentRegion(
+        RocalContext p_context,
+        RocalTensor p_input,
+        bool is_output,
+        float cutoff_db,
+        float reference_power,
+        int reset_interval,
+        int window_length) {
+    Tensor* output1 = nullptr;
+    Tensor* output2 = nullptr;
+    if ((p_context == nullptr) || (p_input == nullptr))
+        ERR("Invalid ROCAL context or invalid input tensor")
+    TensorList output_tensors;
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
+    try {
+        RocalTensorDataType tensor_data_type = RocalTensorDataType::FP32;   // TODO - Change to INT32
+        unsigned number_of_dims = 4;
+        std::vector<size_t> dims1(number_of_dims, 1);
+        dims1.at(0) = context->user_batch_size();
+        auto info1 = TensorInfo(std::vector<size_t>(std::move(dims1)),
+                               context->master_graph->mem_type(),
+                               tensor_data_type);
+        std::vector<size_t> dims2(number_of_dims, 1);
+        dims2.at(0) = context->user_batch_size();
+        auto info2 = TensorInfo(std::vector<size_t>(std::move(dims2)),
+                               context->master_graph->mem_type(),
+                               tensor_data_type);
+        output1 = context->master_graph->create_tensor(info1, is_output);
+        output2 = context->master_graph->create_tensor(info2, is_output);
+        output1->reset_tensor_roi();
+        output2->reset_tensor_roi();
+        output_tensors.push_back(output1);
+        output_tensors.push_back(output2);
+        context->master_graph->add_node<NonSilentRegionNode>({input}, {output1, output2})->init(cutoff_db, reference_power, window_length, reset_interval);
+    } catch(const std::exception& e) {
+        context->capture_error(e.what());
+        ERR(e.what())
+    }
+
+    return std::make_pair(output_tensors.at(0), output_tensors.at(1));
+}
+
+RocalTensor ROCAL_API_CALL
+rocalSpectrogram(
+        RocalContext p_context,
+        RocalTensor p_input,
+        bool is_output,
+        std::vector<float> &window_fn,
+        bool center_windows,
+        bool reflect_padding,
+        RocalSpectrogramLayout spectrogram_layout,
+        int power,
+        int nfft,
+        int window_length,
+        int window_step,
+        RocalTensorOutputType output_datatype) {
+    Tensor* output = nullptr;
+    if ((p_context == nullptr) || (p_input == nullptr)) {
+        ERR("Invalid ROCAL context or invalid input tensor")
+        return output;
+    }
+    auto context = static_cast<Context*>(p_context);
+    auto input = static_cast<Tensor*>(p_input);
+    try {
+        RocalTensorDataType op_tensor_data_type = (RocalTensorDataType)output_datatype;
+        std::vector<size_t> max_dims = input->info().max_shape();
+        TensorInfo output_info = input->info();
+        int window_offset = 0;
+        if(!center_windows)
+            window_offset = window_length;
```

Comment:
```suggestion
        int window_offset = 0;
        if(!center_windows)
            window_offset = window_length;
```

These 3 statements can be combined into 1 line if ternary operator